compiler/p/codegen/OMRCodeGenerator.cpp

```diff
@@ -3239,7 +3239,15 @@ OMR::Power::CodeGenerator::fixedLoadLabelAddressIntoReg(
          self()->itemTracking(offset, label);
          if (offset<LOWER_IMMED||offset>UPPER_IMMED)
             {
-            generateTrg1Src1ImmInstruction(self(), TR::InstOpCode::addis, node, trgReg, self()->getTOCBaseRegister(), self()->hiValue(offset));
+            if (0x00008000 == self()->hiValue(offset))
+               {
+               generateTrg1Src1ImmInstruction(self(), TR::InstOpCode::addis, node, trgReg, self()->getTOCBaseRegister(), 0x7FFF);
+               generateTrg1Src1ImmInstruction(self(), TR::InstOpCode::addis, node, trgReg, trgReg, 0x1);
+               }
+            else
+               {
+               generateTrg1Src1ImmInstruction(self(), TR::InstOpCode::addis, node, trgReg, self()->getTOCBaseRegister(), self()->hiValue(offset));
+               }
```

Comment:
I'm reluctant to have special code added to handle these large offsets for pTOC loads. As @zl-wang pointed out, pTOC loads with offsets this large cannot actually exist, so this new code is effectively dead and cannot be tested.

I'm going to ask a bit of a fundamental question here: does this really need to be fixed at all for pTOC loads? Truncating the value is definitely wrong since it might silently do the wrong thing if these circumstances _do_ arise unintentionally, but if we don't expect pTOC offsets this large, then triggering an assert failure if that does happen is perfectly fine IMO... Maybe a better solution would be to add asserts to these locations themselves that provide a better description of what happened?

Reply:
I'm fine with changing this to an assert of some sort since it is unlikely that an offset of that size is legitimate.

Reply:
I've change this location and other locations that try to load from a large offset from the TOC to an assert.

One key thing to note is I did not change the check in `loadaddrEvaluator`. That one was not a load from the TOC so I wasn't sure if it was impossible to have an offset of `0x7FFF8000` or larger at that point.